lib/jekyll/post.rb

```diff
@@ -118,6 +119,14 @@ def title
       data.fetch('title') { titleized_slug }
     end
 
+    # Public: the Post excerpt_separator, from the YAML Front-Matter or site default
+    #         excerpt_separator value
+    #
+    # Returns the post excerpt_separator
+    def excerpt_separator
+      data.fetch('excerpt_separator') { site.config['excerpt_separator'].to_s }
```

Comment:
Whatever the solution, let's ensure this method always returns a string. Otherwise it'll blow up `generate_excerpt?`